src/test/scala/com.snowplowanalytics.snowplow.badrows/PayloadSpec.scala

```diff
@@ -14,16 +14,25 @@
 package com.snowplowanalytics.snowplow.badrows
 
 import cats.syntax.option._
+
 import io.circe.literal._
 import io.circe.syntax._
+import io.circe.parser.decode
+
 import org.specs2.Specification
+import org.specs2.ScalaCheck
+
+import org.scalacheck.Prop.forAll
 
-class PayloadSpec extends Specification {
+import org.scalacheck.{Arbitrary, Gen}
+
+class PayloadSpec extends Specification with ScalaCheck {
   import Payload._
   def is = s2"""
   encode RawPayload $e1
   encode CollectorPayload $e2
-  encode PartiallyEnrichedEvent $e3
+  encode EnrichmentPayload $e3
+  decode EnrichmentPayload with $$.payload.raw.parameters as object (1-0-0) $e4
```

Comment:
Just in case - there's also a tool helping to write property-based tests supporting removed classes: https://github.com/circe/circe-golden, but this approach is also fine given a time pressure.

Reply:
I was not aware of that, thanks for the info, that matches exactly our use case. If time allows before R119 goes out, I'll update the tests to use it.